pgtrigger/core.py

```diff
@@ -820,140 +867,175 @@ def get(*uris):
 
         return [registry[uri] for uri in uris]
     else:
-        return list(registry.values())
+        return [
+            (model, trigger)
+            for model, trigger in registry.values()
+            if _get_database(model) in databases
+        ]
 
 
-def install(*uris):
+def install(*uris, database=None):
     """
     Install registered triggers matching URIs or all triggers if URIs aren't
     provided. If URIs aren't provided, prune any orphaned triggers from the
-    database
+    database. If a database is provided, will only install triggers
+    registered to a particular database.
     """
     if uris:
-        model_triggers = get(*uris)
+        model_triggers = get(*uris, database=database)
     else:
         model_triggers = [
             (model, trigger)
-            for model, trigger in get()
+            for model, trigger in get(database=database)
             if trigger.get_installation_status(model)[0] != INSTALLED
         ]
 
     for model, trigger in model_triggers:
         LOGGER.info(
-            f'pgtrigger: Installing "{trigger}" trigger'
-            f' for {model._meta.db_table} table.'
+            f'pgtrigger: Installing {trigger} trigger'
+            f' for {model._meta.db_table} table'
+            f' on {_get_database(database)} database.'
         )
         trigger.install(model)
 
     if not uris:  # pragma: no branch
+        # Prune triggers across all databases by default. If a trigger is
+        # configured to execute on a different database, this will help
+        # ensure the old one is dropped
         prune()
 
 
-def get_prune_list():
-    """Return triggers that will be pruned upon next full install"""
+def get_prune_list(database=None):
+    """Return triggers that will be pruned upon next full install
+
+    Args:
+        database (str, default=None): Only return results from this
+            database. Defaults to returning results from all databases
+    """
     installed = {
         (model._meta.db_table, trigger.get_pgid(model))
-        for model, trigger in get()
+        for model, trigger in get(database=database)
     }
 
-    with connection.cursor() as cursor:
-        cursor.execute(
-            'SELECT tgrelid::regclass, tgname, tgenabled'
-            '    FROM pg_trigger'
-            '    WHERE tgname LIKE \'pgtrigger_%\''
-        )
-        triggers = set(cursor.fetchall())
+    if isinstance(database, str):
+        databases = [database]
+    else:
+        databases = database or settings.DATABASES
```

Comment:
Say - could `database` not be a singular object here?
I.e. should this be:
```
elif database is None:
    databases = settings.DATABASES
```

I perhaps haven't fully grokked the possible values for `database`.

Reply:
I'm only handling the case of:

1. A single string (e.g. "default")
2. An iterable of strings

When a false-y object is provided, it defaults to all databases (i.e the dictionary keys of the DATABASES setting)

None of the management commands allow providing multiple databases (yet), but I went ahead and just made all of these functions work with singular or multiple databases if we extend the management commands to take multiple ``--database`` arguments.

If you passed in a random object or iterable of non-strings, it would just fail unexpectedly.

Reply:
Ah doi - the first `if isinstance(database, str):` takes care of the singular case anyway 🤦